app/models/observation_accuracy_experiment.rb

```diff
@@ -586,16 +431,224 @@ def get_sample_for_user_id( user_id )
     )
   end
 
-  def get_incorrect_observations_in_sample
-    samples = ObservationAccuracySample.where( observation_accuracy_experiment_id: id, correct: 0 )
-    return nil unless samples.count.positive?
+  def get_top_level_stats( subset )
+    subset_filter = case subset
+    when "verifiable_results" then "( quality_grade = 'research' OR quality_grade = 'needs_id' )"
+    when "all_results" then nil
+    else "quality_grade = 'research'"
+    end
 
-    oids = samples.map( &:observation_id ).join( "," )
-    FakeView.identify_observations_url(
-      place_id: "any",
-      reviewed: "any",
-      quality_grade: "needs_id,research,casual",
-      id: oids
-    )
+    counts = {
+      incorrect: observation_accuracy_samples.
+        where( [subset_filter, "correct = 0"].compact.join( " AND " ).to_s ).count,
+      uncertain: observation_accuracy_samples.
+        where( [subset_filter, "( correct IS NULL OR correct = -1 )"].compact.join( " AND " ).to_s ).count,
+      correct: observation_accuracy_samples.
+        where( [subset_filter, "correct = 1"].compact.join( " AND " ).to_s ).count
+    }
+    total = counts.values.sum
+    stats = counts.transform_values {| count | ( total.zero? ? 0 : count / total.to_f * 100 ).round( 2 ) }
+
+    descendant_counts = observation_accuracy_samples.where( subset_filter.to_s ).map( &:descendant_count )
+    mean_precision = if descendant_counts.empty?
+      0
+    else
+      descendant_counts.map {| descendant_count | calculate_precision( descendant_count ) }.
+        sum / descendant_counts.count
+    end
+
+    {
+      correct: stats[:correct],
+      uncertain: stats[:uncertain],
+      incorrect: stats[:incorrect],
+      precision: ( mean_precision * 100 ).round( 2 ),
+      sample_size: descendant_counts.count
+    }
+  end
+
+  def taxon_rank_level_categories( key )
+    case key
+    when 5 then "subspecies"
+    when 10 then "species"
+    when 11..20 then "sp.-gen."
+    when 21..30 then "gen.-fam."
+    when 31..40 then "fam.-ord."
+    when 41..50 then "ord.-class"
+    when 51..60 then "class-phy."
+    when 61..70 then "phy.-king."
+    else "life"
+    end
+  end
+
+  def taxon_observations_count_categories( key )
+    case key
+    when 1..100 then "1-100"
+    when 101..1_000 then "100-1k"
+    when 1_001..10_000 then "1k-10k"
+    when 10_001..100_000 then "10k-100k"
+    else ">100k"
+    end
+  end
+
+  def normalize_counts( counts )
+    total = counts.values.flatten.count
+    stats = counts.transform_values do | ids |
+      norm = ( total.zero? ? 0 : ids.count / total.to_f * 100 ).round( 2 )
+      {
+        url: FakeView.observations_url( verifiable: "any", place_id: "any", id: ids.join( "," ) ),
```

Comment:
Not a big deal, but FakeView is a bit of a hack that if we can avoid would be better. Instead of having the model generate the URL, it could return an array of observation_ids which could then be converted to a URL in the view which has access to URL helpers